burr/integrations/opentelemetry.py

```diff
@@ -511,6 +515,91 @@ def initialize_tracer():
+    "anthropic": ("anthropic", "opentelemetry.instrumentation.anthropic", "AnthropicInstrumentor"),
+    "cohere": ("cohere", "opentelemetry.instrumentation.cohere", "CohereInstrumentor"),
+    "google_generativeai": (
+        "google.generativeai",
+        "opentelemetry.instrumentation.google_generativeai",
+        "GoogleGenerativeAiInstrumentor",
+    ),
+    "mistral": ("mistralai", "opentelemetry.instrumentation.mistralai", "MistralAiInstrumentor"),
+    "ollama": ("ollama", "opentelemetry.instrumentation.ollama", "OllamaInstrumentor"),
+    "transformers": (
+        "transformers",
+        "opentelemetry.instrumentation.transformers",
+        "TransformersInstrumentor",
+    ),
+    "together": ("together", "opentelemetry.instrumentation.together", "TogetherAiInstrumentor"),
+    "bedrock": ("bedrock", "opentelemetry.instrumentation.bedrock", "BedrockInstrumentor"),
+    "replicate": ("replicate", "opentelemetry.instrumentation.replicate", "ReplicateInstrumentor"),
+    "vertexai": ("vertexai", "opentelemetry.instrumentation.vertexai", "VertexAIInstrumentor"),
+    "groq": ("groq", "opentelemetry.instrumentation.groq", "GroqInstrumentor"),
+    "watsonx": ("ibm-watsonx-ai", "opentelemetry.instrumentation.watsonx", "WatsonxInstrumentor"),
+    "alephalpha": (
+        "aleph_alpha_client",
+        "opentelemetry.instrumentation.alephalpha",
+        "AlephAlphaInstrumentor",
+    ),
+    "pinecone": ("pinecone", "opentelemetry.instrumentation.pinecone", "PineconeInstrumentor"),
+    "qdrant": ("qdrant_client", "opentelemetry.instrumentation.qdrant", "QdrantInstrumentor"),
+    "chroma": ("chromadb", "opentelemetry.instrumentation.chromadb", "ChromaInstrumentor"),
+    "milvus": ("pymilvus", "opentelemetry.instrumentation.milvus", "MilvusInstrumentor"),
+    "weaviate": ("weaviate", "opentelemetry.instrumentation.weaviate", "WeaviateInstrumentor"),
+    "lancedb": ("lancedb", "opentelemetry.instrumentation.lancedb", "LanceInstrumentor"),
+    "marqo": ("marqo", "opentelemetry.instrumentation.marqo", "MarqoInstrumentor"),
+    "redis": ("redis", "opentelemetry.instrumentation.redis", "RedisInstrumentor"),
+    "langchain": ("langchain", "opentelemetry.instrumentation.langchain", "LangchainInstrumentor"),
+    "llama_index": (
+        "llama_index",
+        "opentelemetry.instrumentation.llamaindex",
+        "LlamaIndexInstrumentor",
+    ),
+    "haystack": ("haystack", "opentelemetry.instrumentation.haystack", "HaystackInstrumentor"),
+    "requests": ("requests", "opentelemetry.instrumentation.requests", "RequestsInstrumentor"),
+    "httpx": ("httpx", "opentelemetry.instrumentation.httpx", "HTTPXClientInstrumentor"),
+    "urllib": ("urllib", "opentelemetry.instrumentation.urllib", "URLLibInstrumentor"),
+    "urllib3": ("urllib3", "opentelemetry.instrumentation.urllib3", "URLLib3Instrumentor"),
+}
+
+INSTRUMENTS = Literal[tuple(INSTRUMENTS_SPECS)]
```

Comment:
IDE completion won't be smart enough for this i think. Better to just hardcode? You can have an assert statement that they're the same length or a test.